w3af/plugins/crawl/spider_man.py

```diff
@@ -185,8 +185,15 @@ def do_ALL(self):
             om.out.information(
                 'The user is navigating through the spider_man proxy.')
 
+        # convert relative URL to absolute if request came from CONNECT
+        if hasattr(self.server, 'chainedHandler'):
+            base_path = "https://" + self.server.chainedHandler.path
+            path = base_path + self.path
+        else:
+            path = self.path
+
         # Convert to url_object
-        path = URL(self.path)
+        path = URL(path)
```

Comment:
This fix doesn't seem to be related with the follow_redirects, could you explain what it fixes?